### PR TITLE
Drop explicit support for python-2.7

### PR DESCRIPTION
**Pull request by alurban**
This PR drops all support for python-2.7, now that we've been running stably in python-3.7:

* Remove imports from `__future__`
* Replace imports from `six` with their python3 equivalents
* Enforce system python versions >= 3.5

This is part of an extensive purge of python2 across our detchar software stack, see gwdetchar/gwdetchar#324 and gwpy/gwsumm#267.

cc @duncanmmacleod, @jrsmith02

### CODE REVIEW

**Comment by alurban** [comment]
@jrsmith02, if it's alright with you, I think we should punt to @duncanmmacleod for review on this one. As noted in the PR notes above, this is part of a broader purge of python2 support across gwpy, gwdetchar, hveto, and gwsumm.

**Comment by jrsmith02** [comment]
I agree!

File: versioneer.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 * https://github.com/warner/python-versioneer
 * Brian Warner
 * License: Public Domain
-* Compatible With: python2.6, 2.7, 3.3, 3.4, 3.5, and pypy
+* Compatible With: python3.5+, conda, and pypy
```
Comment (duncanmmacleod) [inline versioneer.py]: Please undo this change, since we aren't modifying version ever itself in this PR.
```suggestion
* Compatible With: python2.6, 2.7, 3.3, 3.4, 3.5, and pypy
```

File: setup.py [hunk 2]
```diff
@@ -20,10 +20,15 @@
...
 
+import sys
 from setuptools import setup
 
 import versioneer
 
+# enforce python version
+if sys.version < '3.5':
+    raise ImportError("Python versions older than 3.5 are not supported.")
+
```
Comment (duncanmmacleod) [inline setup.py]: I would remove this block, the classifiers and `python_requires` statements are enough.

File: bin/hveto [hunk 3]
```diff
@@ -39,11 +37,7 @@ from getpass import getuser
 from distutils.spawn import find_executable
 from pathlib import Path
 
-try:
-    import configparser
-except ImportError:  # python 2.x
-    import ConfigParser as configparser
-
+import configparser
```
Comment (duncanmmacleod) [inline bin/hveto]: Can you move this up to sit with the rest of the standard library imports?

Comment (alurban) [reply bin/hveto]: @duncanmmacleod, this is done in the latest commit.

**Comment by alurban** [comment]
Thanks for reviewing @duncanmmacleod, I've addressed your comments, all of which were good points

File: bin/hveto [hunk 4]
```diff
@@ -20,37 +20,31 @@
...
-    import ConfigParser as configparser
 
 import numpy
 
 from matplotlib import use
 use('agg')
 
-from gwpy.io.cache import read_cache
 from gwpy.time import to_gps
+from gwpy.io.cache import read_cache
```
Comment (duncanmmacleod) [inline bin/hveto]: Why was this moved?

Comment (alurban) [reply bin/hveto]: @duncanmmacleod, mostly it's because I'm obsessive about the flow of text width in import statements -- I like to go from narrow to wide, for cleanliness of presentation

Comment (duncanmmacleod) [reply bin/hveto]: I prefer to go alphabetically, but I don't really care, just was curious since this isn't strictly python3-related.

**Comment by alurban** [comment]
Thank you @duncanmmacleod!
